### ISSUE TITLE
selfsupervised_learning.py doesn't appear to be self-supervised. It uses labels to calculate the loss.

### ISSUE THREAD

**Issue by vinsis** [k-han/AutoNovel#1]
Hi Kai Han, the classifier model in `selfsupervised_learning.py` [uses cross entropy loss between predicted and actual labels](https://github.com/k-han/AutoNovel/blob/master/selfsupervised_learning.py#L62-L66). The `GenericDataset` used [returns the image and the label](https://github.com/k-han/AutoNovel/blob/master/data/rotationloader.py#L90).

However the paper says and the name suggests that labels are not used for training. Am I interpreting something incorrectly or is the implementation actually not self-supervised? Thanks.

**Comment by k-han** [comment]
Hi, we use the RotNet as our self-supervised model, which learns to predict the rotation applied to an image. Cross entropy loss is used between the predicted rotation and the GT rotation (which is randomly generated). The rotation labels are fetched by this line of the code: https://github.com/k-han/AutoNovel/blob/5eda7e45898cf3fbcde4c34b9c14c743082abd94/data/rotationloader.py#L163  Note that class labels of the images are NOT used at all in self-supervised learning. You may look into Sec. 2.1 in our paper, or the [RotNet paper](https://arxiv.org/pdf/1803.07728.pdf) by Gidaris et al. to learn more about the rotation based self-supervised learning.

**Comment by vinsis** [comment]
Thanks. My bad for looking into it in greater detail.
